cmd/zed/agents/zfs_retire.c

```diff
@@ -165,6 +178,46 @@ find_by_guid(libzfs_handle_t *zhdl, uint64_t pool_guid, uint64_t vdev_guid,
 	return (zhp);
 }
 
+
+
+/*
+ * Take a spare attempt that failed with EBUSY and add it to
+ * the end of a list to try again later.
+ */
+static void
+save_deferred_spare(fmd_hdl_t *hdl, zpool_handle_t *zhp, nvlist_t *vdev)
+{
+	deferred_spare_t *deferred_spare;
+
+	deferred_spare = fmd_hdl_alloc(hdl, sizeof (deferred_spare_t),
+	    FMD_SLEEP);
+
+	deferred_spare->zds_pool_guid = zpool_get_prop_int(zhp,
+	    ZPOOL_PROP_GUID, NULL);
+	verify(nvlist_lookup_uint64(vdev, ZPOOL_CONFIG_GUID,
+	    &(deferred_spare->zds_vdev_guid)) == 0);
+
+	fmd_hdl_debug(hdl, "Saved request pool_guid %llu vdev_guid %llu.",
+	    deferred_spare->zds_pool_guid, deferred_spare->zds_vdev_guid);
+
+	list_insert_tail(&deferred_spares, deferred_spare);
+}
+
+static int
```

Comment:
nit: you could use `boolean_t` for the return type and return `B_TRUE` or `B_FALSE`.  Although I'm fine with it either way.